### DISCUSSION TITLE
Old TODOs tracker

### DISCUSSION THREAD

**Discussion by drHuangMHT** [libp2p/rust-libp2p#4136] [General]
Below are TODOs that were created at least 4 years ago:
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/core/src/transport/timeout.rs#L150-L151
   - rust-lang/rust/issues/34511 has been replaced with rust-lang/rust/issues/63066
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/core/src/upgrade/transfer.rs#L93-L94
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/core/src/upgrade/transfer.rs#L149
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/misc/multistream-select/src/dialer_select.rs#L70-L72
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/misc/multistream-select/src/listener_select.rs#L75-L78
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/floodsub/src/layer.rs#L52-L55
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/behaviour.rs#L2397-L2401
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/handler.rs#L122-L124
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/handler.rs#L1133-L1135
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/protocol.rs#L101-L103 
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/protocol.rs#L136-L138
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/protocol.rs#L454-L456
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/protocol.rs#L624
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/lib.rs#L36-L37
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/kad/src/behaviour/test.rs#L806-L807
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/mdns/src/behaviour/iface/dns.rs#L52
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/mdns/src/behaviour/iface/dns.rs#L67
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/protocols/mdns/src/behaviour/iface/dns.rs#L438
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/swarm/src/lib.rs#L1330
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/swarm/src/handler/one_shot.rs#L32
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/swarm/tests/swarm_derive.rs#L35 
- [ ] https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/transports/uds/src/lib.rs#L327 https://github.com/libp2p/rust-libp2p/blob/7eeb144a73bf6c17fc0a19f463fb5aacc11812be/transports/uds/src/lib.rs#L336

**Comment by thomaseizinger** [comment]
Awesome! Thank you for doing this!

Would you mind linking to the code directly to make the discussion easier?

**Comment by thomaseizinger** [comment]
Thank you for tracking these! Sorry for the late response, this kind of fell-through.

Would you be interested in tackling some of these? I think any form of commented out code can simply be deleted. If we didn't need it in 4 years, we don't need it now :D

Any comment in the form of "it would be nice if" can also be deleted.
